test/process-env-port.test.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const test = require('tap').test;
+const request = require('request');
+const spawn = require('child_process').spawn;
+
+function getRandomInt(min, max) {
+  return Math.floor((Math.random() * ((max - min) + 1))) + min;
+}
+
+const sanePort = getRandomInt(1025, 65536);
+const floatingPointPort = 9090.86;
+const insanePorts = [-1, 65537];
```

Comment:
`-1` is a good addition, but I think we're better served by adding back `Infinity` and `'wow'` to the tests, since they capture additional edge cases

Reply:
I wasn't able to make those work :( The ecstatic behavior under test was that invalid ports are silently ignored and default port is used.  `http-server` validates the input and dies with an exception instead. I played with some options of checking `stderr` or return code but for whatever reason those did not work for 'wow' and Infinity. I'm not sure how valuable it is to test that a process dies anyway, so I kept just the two usecases that worked.

So if we'd implement `ecstatic` behavior and just ignore invalid ports, one could easily put back the original test.

Reply:
@thornjad I think this is the only unresolved point now and I'm not sure if it's worth improving.

Reply:
Yeah, since http-server is meant to be used by CLI anyway (as opposed to as a library like ecstatic), I'm alright with leaving this as you've done it.